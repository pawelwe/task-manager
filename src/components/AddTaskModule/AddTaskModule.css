.AddTaskModule {
  padding-top: 1.2rem;
  border-top: 1px solid #dddddd; }
  .AddTaskModule_btn {
    font-size: 1.4rem;
    font-weight: bold;
    color: crimson;
    background-color: transparent; }
  .AddTaskModule_input {
    display: block;
    width: 100%;
    background-color: #ffffff;
    margin-right: 1rem;
    font-size: 1.2rem;
    font-weight: 100;
    padding: 0.6rem 1rem;
    border: 1px solid #dddddd;
    border-bottom: 0; }
    .AddTaskModule_input::placeholder {
      color: #999999; }
    .AddTaskModule_input:first-of-type {
      margin-top: 1rem; }
    .AddTaskModule_input:last-of-type {
      border-bottom: 1px solid #dddddd;
      margin-bottom: 1rem; }

@media (min-width: 768px) {
  .AddTaskModule {
    margin: 0 2rem; }
    .AddTaskModule_input {
      max-width: calc(33% - 20px); } }

@media (min-width: 1024px) {
  .AddTaskModule {
    margin: 0 3rem; }
    .AddTaskModule_input {
      max-width: calc(25% - 40px); } }
