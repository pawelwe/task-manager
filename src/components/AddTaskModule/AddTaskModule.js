import React from 'react';
import classes from './AddTaskModule.css';

const AddTaskModule = ({ addTaskModule }) => {
  let input, confirm;
  return (
    <footer className={`${classes.AddTaskModule}`}>
      <input
        autoFocus
        placeholder="Module name..."
        className={classes.AddTaskModule_input}
        ref={node => (input = node)}
        onKeyPress={e => {
          if (e.key === 'Enter') {
            confirm.click();
          }
        }}
      />
      <button
        id="addTaskModule"
        ref={node => (confirm = node)}
        onClick={() => {
          addTaskModule(input.value);
          input.value = '';
        }}
        className={classes.AddTaskModule_btn}
      >
        + Add New Tasks Module
      </button>
    </footer>
  );
};

export default AddTaskModule;
