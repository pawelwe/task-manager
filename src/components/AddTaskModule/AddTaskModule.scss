@import '../App/scss/variables';

.AddTaskModule {
  padding-top: 1.2rem;
  border-top: 1px solid $medium-gray-3;

  &_btn {
    font-size: 1.4rem;
    font-weight: bold;
    color: $accent;
    background-color: transparent;
  }
  &_input {
    display: block;
    width: 100%;
    background-color: $white;
    margin-right: 1rem;
    font-size: 1.2rem;
    padding: 0.6rem 1rem;
    border: 1px solid $medium-gray-3;
    border-bottom: 0;
    &::placeholder {
      color: $medium-gray;
    }
    &:first-of-type {
      margin-top: 1rem;
    }
    &:last-of-type {
      border-bottom: 1px solid $medium-gray-3;
      margin-bottom: 1rem;
    }
  }
}

@media (min-width: $tablet) {
  .AddTaskModule {
    margin: 0 2rem;

    &_input {
      max-width: calc(33% - 20px);
    }
  }
}

@media (min-width: $desktop) {
  .AddTaskModule {
    margin: 0 3rem;

    &_input {
      max-width: calc(25% - 40px);
    }
  }
}