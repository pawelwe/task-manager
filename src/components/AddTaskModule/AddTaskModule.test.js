import React from 'react';
import { mount } from 'enzyme';
import AddTaskModule from './AddTaskModule';

describe('AddTaskModule Component', () => {
  const addTaskModule = jest.fn();
  const wrapper = mount(<AddTaskModule addTaskModule={addTaskModule} />);

  it('Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('addTaskModule called', () => {
    wrapper.find('#addTaskModule').simulate('click');
    expect(addTaskModule).toHaveBeenCalled();
  });
});
