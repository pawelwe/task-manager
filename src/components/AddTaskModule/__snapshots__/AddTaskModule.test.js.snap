// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTaskModule Component Snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <AddTaskModule
    addTaskModule={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "addTaskModule": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": <footer
        class="undefined"
>
        <input
                placeholder="Module name..."
        />
        <button
                id="addTaskModule"
        >
                + Add New Tasks Module
        </button>
</footer>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            autoFocus={true}
            className={undefined}
            onKeyPress={[Function]}
            placeholder="Module name..."
/>,
          <button
            className={undefined}
            id="addTaskModule"
            onClick={[Function]}
>
            + Add New Tasks Module
</button>,
        ],
        "className": "undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <input
            placeholder="Module name..."
/>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoFocus": true,
            "className": undefined,
            "onKeyPress": [Function],
            "placeholder": "Module name...",
          },
          "ref": [Function],
          "rendered": Array [
            undefined,
          ],
          "type": "input",
        },
        Object {
          "instance": <button
            id="addTaskModule"
>
            + Add New Tasks Module
</button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+ Add New Tasks Module",
            "className": undefined,
            "id": "addTaskModule",
            "onClick": [Function],
          },
          "ref": [Function],
          "rendered": Array [
            "+ Add New Tasks Module",
          ],
          "type": "button",
        },
      ],
      "type": "footer",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "addTaskModule": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <footer
          class="undefined"
>
          <input
                    placeholder="Module name..."
          />
          <button
                    id="addTaskModule"
          >
                    + Add New Tasks Module
          </button>
</footer>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              autoFocus={true}
              className={undefined}
              onKeyPress={[Function]}
              placeholder="Module name..."
/>,
            <button
              className={undefined}
              id="addTaskModule"
              onClick={[Function]}
>
              + Add New Tasks Module
</button>,
          ],
          "className": "undefined",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <input
              placeholder="Module name..."
/>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoFocus": true,
              "className": undefined,
              "onKeyPress": [Function],
              "placeholder": "Module name...",
            },
            "ref": [Function],
            "rendered": Array [
              undefined,
            ],
            "type": "input",
          },
          Object {
            "instance": <button
              id="addTaskModule"
>
              + Add New Tasks Module
</button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "+ Add New Tasks Module",
              "className": undefined,
              "id": "addTaskModule",
              "onClick": [Function],
            },
            "ref": [Function],
            "rendered": Array [
              "+ Add New Tasks Module",
            ],
            "type": "button",
          },
        ],
        "type": "footer",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
